components/nexus-core/src/main/java/org/sonatype/nexus/internal/blobstore/BlobStoreManagerImpl.java

```diff
@@ -245,22 +220,18 @@ private void untrack(final String name) {
   }
 
   /**
-   * Ensure that the path is not already used by another BlobStore
+   * Ensure that the configuration attributes are not already used by another BlobStore
    */
   private void validate(final BlobStoreConfiguration configuration) {
-    final String path = getPath(configuration.getAttributes());
-    boolean duplicatePath = Iterables.any(store.list(), new Predicate<BlobStoreConfiguration>()
-    {
-      @Override
-      public boolean apply(final BlobStoreConfiguration input) {
-        return path.equals(getPath(input.getAttributes()));
+    for (BlobStoreConfiguration blobStoreConfiguration : store.list()) {
+      if (Maps.difference(blobStoreConfiguration.getAttributes(), configuration.getAttributes()).areEqual()) {
```

Comment:
Once we have other types, will probably have to inject something like Map<String, Validator> corresponding to the blobStorePrototypes, signature like:
```
void validate(BlobStoreConfiguration configuration, List<BlobStoreConfiguration> existingConfigurations)
```